doc/source/internals/ondisk/objects.rst

```diff
@@ -0,0 +1,302 @@
+
+:ref:`schema-serialization`, when performed, will make sure the attribute order is interpreted correctly even
+across varying versions of the schema. It also persists serialization options, like alignment size and
+other such configurable parameters.
+
+Fixed size attributes include the primitive types: numbers (integer and float) and bool.
+Non fixed-size attributes are encoded as a 64-bit pointer relative to the offset where the object header
+is located.
+
+Only attributes set and not None will be included in the attribute data, though. An attribute is set
+if ``has_bitmap & (1 << attribute_position) != 0``. Similarly, an attribute is None if
+``none_bitmap & (1 << attribute_position) != 0``, where ``attribute_position`` depends on the attribute
+order computed by the rule mentioned above (or serialized in the persistent schema definition when
+:ref:`schema-serialization` is performed).
+
+After that data, padding is added to reach a properly aligned location, and then variable-sized attribute
+data follows. Variable-sized attributes are appended after padding without any header, in arbitrary order.
+They can be located through the relative pointers present in the attribute data section.
+
+Objects *can* refer, through relative pointers, to any value stored in lower locations, so the
+pointers may contain negative offsets in that case, and are thus signed.
+
+.. _rtti-wrapping:
+
+RTTI wrapping
+-------------
+
+Objects can be contained within objects, and sometimes the type (:class:`~sharedbuffers.mapped_struct.Schema`) of a
+reference isn't known in advance. This happens for all container types when the type of the contents isn't uniform or
+fixed in the :class:`~sharedbuffers.mapped_struct.Schema`, and for attribute references when the referent type is
+annotated plainly as ``object``.
+
+One can produce :term:`RTTI`-wrapped values on demand through `mapped_object.pack_into`, which takes objects of any
+supported type, and produces the structure described in this section.
+
+So, when necessary, values are wrapped in a header with :term:`RTTI`. The process is slightly different for
+variable-sized types vs fixed-size types. In the case of variable-size types, the header is padded
+to reach alignment, and the value is then appended as it normally would if the type were known right after
+the padded header.
+
+For fixed-size types, the header structure and the value itself are merged to achieve a more compact representation:
+
+.. code-block:: C
+
+    struct RTTIHeader {
+        char typecode;
+        union {
+            int8 i8;
+            uint8 ui8;
+            int16 i16;
+            uint16 ui16;
+            int32 i32;
+            uint32 ui32;
+            int64 i64;
+            uint64 ui64;
+            float f;
+            double d;
+            bool b;
+        }
+    }
+
+As can be gleaned from the above, the value itself won't be aligned except for the byte-sized types, and
```

Comment:
This doesn't match the C code. A `union` has the needed size and alignment to hold any of its constituent members.